### PR TITLE
PEP8 style sweep

### PR DESCRIPTION
**Pull request by feygon**
Without affecting test ratios, I have made a sweep of the Arrays
algorithms, and improved style, commentary, whitespace, and some
translations

flatten.py -- added commentary

Garage.py -- added commentary, enforced line width on existing
commentary

Josephus.py -- enforced commentary alignment, binary operator spacing

Limit.py -- enforced binary operator spacing, enforced commentary width
This function is very similar to Python array-slicing, except by value.

longest_non_repeat.py -- enforced spacing style in commentary
added get functions
Enforced function whitespace of one carriage return

merge_intervals.py -- switched docstrings so the subject exposition was
in the intro and the action done by the class was under the class,
cleaned up other docstrings
-- this class does need a docstring, but not the mathematician's
exposition. What does this object do?

n_sum.py -- added whitespace around recursive call, added (TL:DR)
commentary, added docstrings, added commentary identifying recursive
call, enforced line width and pep8 line break style WRT parameters on
newlines w/ parentheses (e.g. lines 81-84, 88-90), enforced spelling
consistency in argument parameters (e.g. target/taget)

plus_one.py -- removed unused variable in plus_one_v3 method, clarified
opening docstring

rotate.py -- added comment line 24 for intentionally unused variable,
enforced whitespace guidelines between methods

three_sum.py -- specified opening docstring to indicate distinct
elements in solution sets

top_1.py -- improved translation in commentary

trimmean.py -- improved translation in commentary, removed leftover
commented instrumentation

### CODE REVIEW

**Comment by feygon** [comment]
Included language inviting contributors to improve documentation.

(Put an `X` inside the `[ ]` to denote check mark `[X]`.)


- [ ] **If creating a new file :**
  - [ ] added links to it in the README files ?
  - [ ] included tests with it ?
  - [ ] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [X] **if done some changes :**
  - [X] wrote short description in the PR explaining what the changes do ?
  - [X] Fixes #[12], #[242] if related to any issue

- [x] **other**

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Shared\\Python36_64\\python.exe"
```
Comment (keon) [inline .vscode/settings.json]: please add .vscode to .gitignore

File: algorithms/arrays/flatten.py [hunk 2]
```diff
@@ -12,9 +12,9 @@ def flatten(input_arr, output_arr=None):
         output_arr = []
     for ele in input_arr:
         if isinstance(ele, Iterable):
-            flatten(ele, output_arr)
+            flatten(ele, output_arr)    #tail-recursion
         else:
-            output_arr.append(ele)
+            output_arr.append(ele)      #produce the result
```
Comment (keon) [inline algorithms/arrays/flatten.py]: Nit:
space after #

File: algorithms/arrays/flatten.py [hunk 3]
```diff
@@ -26,6 +26,6 @@ def flatten_iter(iterable):
     """
     for element in iterable:
         if isinstance(element, Iterable):
-            yield from flatten_iter(element)
+            yield from flatten_iter(element)
```
Comment (keon) [inline algorithms/arrays/flatten.py]: please remove the trailing spaces at the end.
There should be an automatic setting for this.

**Comment by keon** [comment]
Thanks for the contribution!
I gave a brief look, but it looks good to me.
I am adding some minor comments and come back and do a full review today or tomorrow.

**Comment by keon** [comment]
ill merge this PR since it is not affecting logic

**Pull request merged**